### DISCUSSION TITLE
[V5] Change S3 bucket being used in Drive programmatically

### DISCUSSION THREAD

**Discussion by aayush123** [adonisjs/core#3475] [Help]
Hi,

Is there any way to dynamically update the S3 bucket being used in the Drive package?

I understand that I can configure multiple disks in the config file, however, my use-case is that I'm building a micro service that interacts with AWS S3.
The entire application uses multiple buckets and I would like the S3 service to accept a bucket name as parameter when other services interact with it.
This is being done so that S3 has a single point of access and to improve security.

Any help is much appreciated.
Thanks!

**Comment by Julien-R44** [comment]
Unfortunately this does not seem possible with the current exposed API
Why not create as many disks as buckets? This seems to me to be the easiest solution, especially since there is a limit of 100 buckets per account on AWS so you won't have to create thousands of them

Another solution could be to fork drive-s3 and basically make the `config` property public so you can edit it on the fly. It's definitely not clean but it may do the trick 🤷 However at this point, I think it's better to use the AWS S3 SDK directly

**Comment by openstepmedia** [comment]
I haven't tested this.. but have you tried updating your config/drive.ts file:

```
import Env from '@ioc:Adonis/Core/Env'
import { DriveConfig } from '@ioc:Adonis/Core/Drive'
import Application from '@ioc:Adonis/Core/Application'

/*
|--------------------------------------------------------------------------
| Drive Config
|--------------------------------------------------------------------------
|
| The `DriveConfig` relies on the `DisksList` interface which is
| defined inside the `contracts` directory.
|
*/
const driveConfig: DriveConfig = {
  disk: Env.get('DRIVE_DISK'),

  disks: {
    local: {
      driver: 'local',
      visibility: 'public',
      root: Application.tmpPath('uploads'),
      serveFiles: true,
      basePath: '/uploads',
    },

    /*
    |--------------------------------------------------------------------------
    | S3 Driver
    |--------------------------------------------------------------------------
    |
    | Uses the S3 cloud storage to manage files. Make sure to install the s3
    | drive separately when using it.
    |
    |**************************************************************************
    | npm i @adonisjs/drive-s3
    |**************************************************************************
    |
    */
    s3bucketA: {
     driver: 's3',
     visibility: 'private',
     key: Env.get('S3_KEY'),
     secret: Env.get('S3_SECRET'),
     region: Env.get('S3_REGION'),
     bucket: Env.get('S3_BUCKET_A'), <-- Add to .env
     endpoint: Env.get('S3_ENDPOINT'),
    },
	
    s3bucketB: {
     driver: 's3',
     visibility: 'private',
     key: Env.get('S3_KEY'),
     secret: Env.get('S3_SECRET'),
     region: Env.get('S3_REGION'),
     bucket: Env.get('S3_BUCKET_B'),  <-- Add to .env
     endpoint: Env.get('S3_ENDPOINT'),
    },
  },
}

export default driveConfig
```

Then in your code you can do:

```
await Drive.use('s3bucketA').put(location, buffer)
await Drive.use('s3bucketB').put(location, buffer)
```

**Comment by thetutlage** [comment]
Yup, not possible right now. However, I am open for the change, so feel free to make a PR to allow a bucket at runtime.
